syntaxes/src/expression/grammar.ts

```diff
@@ -0,0 +1,886 @@
+            },
+          },
+        },
+      ],
+    },
+
+    parenExpression: {
+      begin: /\(/,
+      beginCaptures: {
+        '0': {
+          name: 'meta.brace.round.ts',
+        },
+      },
+      end: /\)/,
+      endCaptures: {
+        '0': {
+          name: 'meta.brace.round.ts',
+        },
+      },
+      patterns: [
+        {
+          include: '#ngExpression',
+        },
+        {
+          include: '#punctuationComma',
+        },
+      ],
+    },
+
+    punctuationAccessor: {
+      name: 'punctuation.accessor.ts',
+      match: /\?\.|\!\.|\./,
+    },
+
+    punctuationComma: {
+      name: 'punctuation.separator.comma.ts',
+      match: /,/,
+    },
+
+    punctuationSemicolon: {
+      name: 'punctuation.terminator.statement.ts',
+      match: /;/,
+    },
+
+    qstringDouble: {
+      name: 'string.quoted.double.ts',
+      begin: /"/,
+      beginCaptures: {
+        '0': {
+          name: 'punctuation.definition.string.begin.ts',
+        },
+      },
+      end: /(")|((?:[^\\\n])$)/,
+      endCaptures: {
+        '1': {
+          name: 'punctuation.definition.string.end.ts',
+        },
+        '2': {
+          name: 'invalid.illegal.newline.ts',
+        },
+      },
+      patterns: [
+        {
+          include: '#stringCharacterEscape',
+        },
+      ],
+    },
+
+    qstringSingle: {
+      name: 'string.quoted.single.ts',
+      begin: /'/,
+      beginCaptures: {
+        '0': {
+          name: 'punctuation.definition.string.begin.ts',
+        },
+      },
+      end: /(\')|((?:[^\\\n])$)/,
+      endCaptures: {
+        '1': {
+          name: 'punctuation.definition.string.end.ts',
+        },
+        '2': {
+          name: 'invalid.illegal.newline.ts',
+        },
+      },
+      patterns: [
+        {
+          include: '#stringCharacterEscape',
+        },
+      ],
+    },
+
+    stringCharacterEscape: {
+      name: 'constant.character.escape.ts',
+      match: /\\(x\h{2}|[0-2][0-7]{0,2}|3[0-6][0-7]?|37[0-7]?|[4-7][0-7]?|.|$)/,
+    },
+
+    ternaryExpression: {
+      begin: /(?=\?)(?!\?\.)/,
+      end: /(?=$|"|[;,})\]])/,
+      patterns: [
+        {
+          include: '#ternaryOperator',
+        },
+        {
+          include: '#ngExpression',
+        },
+      ],
+    },
+
+    ternaryOperator: {
+      begin: /(\?)(?!\.)/,
+      beginCaptures: {
+        '0': {
+          name: 'keyword.operator.ternary.ts',
+        },
+      },
+      end: /(:)/,
+      endCaptures: {
+        '0': {
+          name: 'keyword.operator.ternary.ts',
+        },
+      },
+      patterns: [
+        {
+          include: '#ngExpression',
+        },
+      ],
+    },
+
+    thisLiteral: {
+      name: 'variable.language.this.ts',
+      match: /(?<!\.|\$)\bthis\b(?!\$)/,
+    },
+
+    tsString: {
+      patterns: [
+        {
+          include: '#qstringSingle',
+        },
+        {
+          include: '#qstringDouble',
+        },
```

Comment:
What about template strings ("`")?

Reply:
@ayazhafiz Are template literals valid in template expressions? I didn't think they were but I could be wrong

Reply:
Angular does not allow lexer characters in the template expression as far as I know... IMO it's up to the template grammar to match them

Reply:
okay good call